### PR TITLE
Readme update to change Node supported versions

### PR DESCRIPTION
**Pull request by CaptFalc**
# Description

I went and updated the supported versions of Node.js from v16 and v18 to v20 to account for updates in Node.js usage

## Checklist

- ~~[ ] I have added before and after screenshots for UI changes~~
- ~~[ ] I have added JSON response output for API changes~~
- ~~[ ] I have added steps to reproduce and test for bug fixes in the description~~
- ~~[ ] I have commented on my code, particularly in hard-to-understand areas~~
- ~~[ ] My changes generate no new warnings~~
- ~~[ ] I have added tests that prove my fix is effective or that my feature works~~

### CODE REVIEW

**Review by cameronperera — APPROVED** [review]
Thanks for catching that!

File: README.md [hunk 1]
```diff
@@ -67,7 +67,7 @@ You will need the following things properly installed on your computer.
 
 * [Git][git]
 * [Node.js][node]
-  * Supported versions: `v16` and `v18`.
+  * Supported versions: `v20`.
```
Comment (calcaide) [inline README.md:70]: We allow users to run v18 and v20, as per [engines](https://github.com/hashicorp/boundary-ui/blob/main/package.json#L82) config. [More info about engines here](https://docs.npmjs.com/cli/v10/configuring-npm/package-json#engines).

Comment (CaptFalc) [reply README.md]: Bet, thanks for the catch Carlos!

Comment (cameronperera) [reply README.md:70]: Oh right. Thanks @calcaide, I forgot we still support 18.

**Review by DhariniJeeva — APPROVED** [review]
awesome catch. congrats on your first PR 🎉

**Review by lisbet-alvarez — APPROVED** [review]
🎊 Awesome job! Congrats on your first PR!! 🙌

**Review by calcaide — APPROVED** [review]
Nice catch!! and congrats on your first PR!!

**Review by ZedLi — APPROVED** [review]
Great catch! 🎉

**Pull request merged**